test/python/synthesis/test_synthesis.py

```diff
@@ -1544,6 +1545,41 @@ def inverse(self, annotated: bool = False):
         circ = decomposer(unitary)
         self.assertEqual(Operator(unitary), Operator(circ))
 
+    def test_assertPickle(self):
+        """Assert that TwoQubitControlledUDecomposer supports pickle"""
+
+        decomp = TwoQubitControlledUDecomposer(RXXGate)
+
+        pkl = pickle.dumps(decomp, protocol=max(4, pickle.DEFAULT_PROTOCOL))
+        decomp_cpy = pickle.loads(pkl)
+        msg_base = f"decomp:\n{decomp}\ndecomp_cpy:\n{decomp_cpy}"
+        self.assertEqual(type(decomp), type(decomp_cpy), msg_base)
+        self.assertEqual(decomp.rxx_equivalent_gate, decomp_cpy.rxx_equivalent_gate, msg=msg_base)
+        self.assertEqual(decomp.euler_basis, decomp_cpy.euler_basis, msg=msg_base)
+
+    def test_assertPickle_with_dill(self):
```

Comment:
Also snake-case  here too:
```suggestion
    def test_assert_pickle_with_dill(self):
```

Reply:
done in [92678d1](https://github.com/Qiskit/qiskit/pull/14038/commits/92678d149a1fc5b480b9d1ecab7d32df4a6cadf4)